cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -246,6 +289,41 @@ public void run() {
     }
   }
 
+  private void runInTransaction(WorkflowAction action, TransactionContext transactionContext) {
```

Comment:
Its better to create the instance of `TransactionContext` in the method instead of passing it as a method parameter, so that we will have all transaction related code localized.